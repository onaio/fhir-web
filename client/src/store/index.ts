// going to use this store initializer instead of the one in
// @opensrp/store package, for some reason its not working
// wehn passed to the redux provider
import { connectReducer, getConnectedStore } from '@onaio/connected-reducer-registry';
import { gateKeeperReducer, gateKeeperReducerName } from '@onaio/gatekeeper';
import reducerRegistry, { combine, Registry } from '@onaio/redux-reducer-registry';
import session, { reducerName as sessionReducer } from '@onaio/session-reducer';
import {
  reducerName as keycloakUsersReducerName,
  reducer as keycloakUsersReducer,
} from '@opensrp/store';
import { Reducer } from 'redux';
import { RouterState } from 'connected-react-router';

/** Initial reducers in the reducer registry */
const defaultReducers: Registry = {
  router: connectReducer as Reducer<RouterState>,
};

/** Add users reducer to registry */
defaultReducers[sessionReducer] = session;
defaultReducers[gateKeeperReducerName] = gateKeeperReducer;
// added this segment because registering from the user management
// package component is failing
defaultReducers[keycloakUsersReducerName] = keycloakUsersReducer;

/** The initial store for the reveal web app */
const store = getConnectedStore(defaultReducers);

/** Set listener to add reducers to store when registered */
reducerRegistry.setChangeListener((reducers) => {
  store.replaceReducer(combine(reducers));
});

export default store;
