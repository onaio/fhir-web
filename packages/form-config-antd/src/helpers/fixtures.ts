/* eslint-disable @typescript-eslint/camelcase */
export const manifestRelease4 = {
  identifier: '1.0.30',
  json: '{"forms_version":"1.0.16","identifiers":["blood_screening.json"}',
  appId: 'org.smartregister.foo',
  appVersion: '1.2.18',
  createdAt: '2020-06-09T14:40:04.189+02:00',
  updatedAt: '2020-06-09T14:40:04.189+02:00',
};

export const fixManifestReleases = [
  {
    identifier: '1.0.11',
    json: '{"forms_version":"1.0.11","identifiers":["blood_screening.json"}',
    appId: 'org.smartregister.reveal',
    appVersion: '1.2.14',
    createdAt: '2020-06-08T15:00:38.413+02:00',
    updatedAt: '2020-06-08T15:00:38.413+02:00',
  },
  {
    identifier: '1.0.12',
    json: '{"forms_version":"1.0.12","identifiers":["blood_screening.json"}',
    appId: 'org.smartregister.reveal',
    appVersion: '1.2.14',
    createdAt: '2020-06-08T15:03:00.950+02:00',
    updatedAt: '2020-06-08T15:03:00.950+02:00',
  },
  {
    identifier: '1.0.16',
    json: '{"forms_version":"1.0.16","identifiers":["blood_screening.json"}',
    appId: 'org.smartregister.reveal',
    appVersion: '1.2.16',
    createdAt: '2020-06-09T14:40:04.189+02:00',
    updatedAt: '2020-06-09T14:40:04.189+02:00',
  },
];

export const manifestFile1 = {
  createdAt: 'Jun 19, 2020, 12:31:37 PM',
  form_relation: '',
  id: '52',
  identifier: 'test-form-1.json',
  isDraft: false,
  isJsonValidator: false,
  jursdiction: '',
  label: 'test form',
  module: '',
  version: '1.0.26',
};

export const manifestFile2 = {
  createdAt: 'Jun 19, 2020, 4:23:22 PM',
  form_relation: '',
  id: '53',
  identifier: 'reveal-test-file.json',
  isDraft: false,
  isJsonValidator: false,
  jursdiction: '',
  label: 'test publish',
  module: '',
  version: '1.0.27',
};

export const manifestFile3 = {
  createdAt: 'Jun 19, 2020, 6:23:22 PM',
  form_relation: '',
  id: '54',
  identifier: 'foo.json',
  isDraft: false,
  isJsonValidator: false,
  jursdiction: '',
  label: 'foo',
  module: 'baz',
  version: '1.0.28',
};

export const draftFile1 = {
  ...manifestFile1,
  isDraft: true,
  isJsonValidator: false,
};

export const draftFile2 = {
  ...manifestFile2,
  isDraft: true,
  isJsonValidator: false,
};

export const draftFile3 = {
  ...manifestFile3,
  isDraft: true,
  isJsonValidator: false,
};

export const fixManifestFiles = [manifestFile1, manifestFile2];

export const FixManifestDraftFiles = [
  {
    ...fixManifestFiles[0],
    isDraft: false,
  },
  {
    ...fixManifestFiles[1],
    isDraft: false,
  },
];

export const downloadFile = {
  clientForm: {
    json: JSON.stringify(fixManifestFiles[0]),
  },
};
