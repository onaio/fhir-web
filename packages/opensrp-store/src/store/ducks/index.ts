/** users ducks modules: actions, actionCreators, reducer and selectors */
import { Dictionary } from '@onaio/utils';
import { keyBy, values } from 'lodash';
import intersect from 'fast_array_intersect';
import { AnyAction, Store } from 'redux';
import { createSelector, OutputParametricSelector } from 'reselect';
import SeamlessImmutable from 'seamless-immutable';

/** The reducer name */
export const reducerName = 'keycloakUsers';

/** Interface for user json object */
export interface KeycloakUser {
  access?: {
    manageGroupMembership: boolean;
    view: boolean;
    mapRoles: boolean;
    impersonate: boolean;
    manage: boolean;
  };
  createdTimestamp?: number;
  disableableCredentialTypes?: string[];
  email?: string;
  emailVerified?: boolean;
  enabled?: boolean;
  firstName: string;
  id: string;
  lastName: string;
  notBefore?: number;
  requiredActions?: string[];
  totp?: boolean;
  username: string;
}

// actions

/** action type for fetching keycloak users */
export const KEYCLOAK_USERS_FETCHED = 'keycloak/reducer/users/userS_FETCHED';
/** action type for removing keycloak users */
export const REMOVE_KEYCLOAK_USERS = 'keycloak/reducer/users/REMOVE_KEYCLOAK_USERS';

/** interface action to add users to store */
export interface FetchKeycloakUsersAction extends AnyAction {
  usersById: Dictionary<KeycloakUser>;
  type: typeof KEYCLOAK_USERS_FETCHED;
}

/** Interface for removeusersAction */
export interface RemoveKeycloakUsersAction extends AnyAction {
  usersById: Dictionary<KeycloakUser>;
  type: typeof REMOVE_KEYCLOAK_USERS;
}

/** Create type for users reducer actions */
export type KeycloakUsersActionTypes =
  | FetchKeycloakUsersAction
  | RemoveKeycloakUsersAction
  | AnyAction;

// action Creators

/**
 * Fetch users action creator
 *
 * @param {Array} usersList - keycloak users
 * @returns {object} - the dispatcher to remove the user
 */
export const fetchKeycloakUsers = (usersList: KeycloakUser[] = []): FetchKeycloakUsersAction => {
  return {
    usersById: keyBy(usersList, (user: KeycloakUser) => user.id),
    type: KEYCLOAK_USERS_FETCHED,
  };
};

/** Remove users action creator
 *
 * @returns {object} - the dispatcher to remove the user
 */
export const removeKeycloakUsers = (): RemoveKeycloakUsersAction => {
  return {
    usersById: {},
    type: REMOVE_KEYCLOAK_USERS,
  };
};

// The reducer

/** interface for keycloak users state in redux store */
export interface KeycloakUsersState {
  usersById: Dictionary<KeycloakUser> | Dictionary;
}

/** Create an immutable keycloak users state */
export type ImmutableKeycloakUsersState = KeycloakUsersState &
  SeamlessImmutable.ImmutableObject<KeycloakUsersState>;

/** initial keycloak users state */
export const initialState: ImmutableKeycloakUsersState = SeamlessImmutable({
  usersById: {},
});

/**
 * the users reducer function
 *
 * @param {object} state - keycloak users states
 * @param {object} action - keycloak users actions
 * @returns {object} - the updated states
 */
export function reducer(
  state: ImmutableKeycloakUsersState = initialState,
  action: KeycloakUsersActionTypes
): ImmutableKeycloakUsersState {
  switch (action.type) {
    case KEYCLOAK_USERS_FETCHED:
      return SeamlessImmutable({
        ...state,
        usersById: { ...state.usersById, ...action.usersById },
      });
    case REMOVE_KEYCLOAK_USERS:
      return SeamlessImmutable({
        ...state,
        usersById: action.usersById,
      });

    default:
      return state;
  }
}

// Selectors

export interface KeycloakUsersFilters {
  id: string[] /** get all users whose ids appear in this array */;
  username?: string /** get users whose username includes text in the username field */;
}

// NON MEMOIZED SELECTORS

/**
 * Gets id from KeycloakUsersFilters
 *
 * @param {any} _ - the redux store
 * @param {Array} props - the keycloak users filters object
 * @returns {string} - the keycloak user id
 */
export const getUserIds = (_: Partial<Store>, props: KeycloakUsersFilters): string[] => props.id;

/**
 * Gets name from KeycloakUsersFilters
 *
 * @param {any} _ - the redux store
 * @param {object} props - the users filters object
 * @returns {string} - the keycloak user name
 */
export const getUsername = (_: Partial<Store>, props: KeycloakUsersFilters): string | undefined =>
  props.username;

/** returns all users in the store as values whose keys are their respective ids
 *
 * @param {any} state - the redux store
 * @returns {object} - users object as values, respective ids as keys
 */
export function getKeycloakUsersById(state: Partial<Store>): { [key: string]: KeycloakUser } {
  return (state as Dictionary)[reducerName].usersById;
}

/** gets keycloak users as an array of user objects
 *
 * @param {any} state - the redux store
 * @returns {Array} - an array of users objs
 */
export function getKeycloakUsersArray(state: Partial<Store>): KeycloakUser[] {
  return values(getKeycloakUsersById(state));
}

/**
 * Gets all practitioners whose name includes phrase given in name filter prop
 *
 * @returns {Array} - practitioners whose name includes phrase given in name filter prop
 */
export const getUsersByUsername = (): OutputParametricSelector<
  Partial<Store>,
  KeycloakUsersFilters,
  KeycloakUser[],
  (res1: KeycloakUser[], res2: string | undefined) => KeycloakUser[]
> =>
  createSelector(getKeycloakUsersArray, getUsername, (usersArray, username) =>
    username
      ? usersArray.filter((user: KeycloakUser) =>
          user.username.toLowerCase().includes(username.toLowerCase())
        )
      : usersArray
  );

// MEMOIZED SELECTORS

/**
 * Gets all users whose identifiers appear in ids filter prop value
 *
 * @returns {Array} - users whose identifiers appear in ids filter prop value
 */
export const getKeycloakUsersByIds = (): OutputParametricSelector<
  Partial<Store>,
  KeycloakUsersFilters,
  KeycloakUser[],
  (
    res1: {
      [key: string]: KeycloakUser;
    },
    res2: string[],
    res3: KeycloakUser[]
  ) => KeycloakUser[]
> =>
  createSelector(
    getKeycloakUsersById,
    getUserIds,
    getKeycloakUsersArray,
    (usersById, ids, usersArray) => {
      if (ids === undefined) {
        return usersArray;
      }
      if (ids.length > 0) {
        return ids.map((id: string) => usersById[id]);
      }
      return [];
    }
  );

/** practitioner array selector factory
 * aggregates response from all applied filters and returns results
 *
 * @returns {Array} - aggregates response from all applied filters and returns results
 */
export const makeKeycloakUsersSelector = (): OutputParametricSelector<
  Partial<Store>,
  KeycloakUsersFilters,
  KeycloakUser[],
  (res1: KeycloakUser[], res2: KeycloakUser[]) => KeycloakUser[]
> =>
  createSelector(getKeycloakUsersByIds(), getUsersByUsername(), (arr1, arr2) => {
    return intersect([arr1, arr2], JSON.stringify);
  });
