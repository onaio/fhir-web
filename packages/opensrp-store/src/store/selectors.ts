import { reducerName as sessionReducerName } from '@onaio/session-reducer';
import { Store } from 'redux';
import { Dictionary } from '@onaio/utils';
import { createSelector } from 'reselect';

/** get API Token from the Redux store
 *
 * @param {any} state - the redux store
 */
interface StoreState {
  session: {
    [key: string]: Dictionary;
  };
}

interface APIFilters {
  accessToken?: boolean;
  apiToken?: boolean;
  providerState?: boolean;
}

/** Fetch access token from store if access_token filter val is true
 *
 * @param {object} _ - redux store
 * @param {object} props - api filters object
 * @returns {boolean} returns boolean value
 */
export const fetchAccessToken = (_: Partial<Store>, props: APIFilters): boolean | null =>
  props.accessToken || null;

/** Fetch api token from store if api_token filter val is true
 *
 * @param {object} _ redux store
 * @param {object} props api filters object
 * @returns {boolean} returns boolean value
 */
export const fetchApiToken = (_: Partial<Store>, props: APIFilters): boolean | null =>
  props.apiToken || null;

/** Fetch oauth provider state from store if providerState filter val is true
 *
 * @param {Object} _ redux store
 * @param {Object} props api filters object
 * @returns {boolean} returns boolean value
 */
export const fetchOauthProviderState = (_: Partial<Store>, props: APIFilters): boolean | null =>
  props.providerState || null;

/** Gets extra data object from store
 *
 * @param {object} state - redux store
 * @returns {Object}
 */

export const getExtraData = (state: Partial<Store>): Dictionary =>
  (state as StoreState)[sessionReducerName].extraData;

/** API state selector
 *
 * @returns {Object} - the states
 */
export const makeAPIStateSelector = () =>
  createSelector(
    [fetchAccessToken, fetchApiToken, fetchOauthProviderState, getExtraData],
    (accessToken, apiToken, oauthState, extraData) => {
      if (accessToken) {
        return (extraData.oAuth2Data && extraData.oAuth2Data.access_token) || null;
      } else if (apiToken) {
        return extraData.api_token || null;
      } else if (oauthState) {
        return (extraData.oAuth2Data && extraData.oAuth2Data.state) || null;
      }
    }
  );
