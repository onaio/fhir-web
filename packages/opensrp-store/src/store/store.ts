import { connectReducer, getConnectedStore } from '@onaio/connected-reducer-registry';
import { gateKeeperReducer, gateKeeperReducerName } from '@onaio/gatekeeper';
import reducerRegistry, { combine, Registry } from '@onaio/redux-reducer-registry';
import session, { reducerName as sessionReducer } from '@onaio/session-reducer';
import { Reducer } from 'redux';
import { RouterState } from 'connected-react-router';
import { Dictionary } from '@onaio/utils';

/** declare globals interface */
declare global {
  interface Window {
    __PRELOADED_STATE__: Dictionary;
  }
}

/** Initial reducers in the reducer registry */
const defaultReducers: Registry = {
  router: connectReducer as Reducer<RouterState>,
};

// Grab the state from a global variable injected into the server-generated HTML
const preloadedState = window.__PRELOADED_STATE__ || {};
// Allow the passed state to be garbage-collected
delete window.__PRELOADED_STATE__;

/** Add users reducer to registry */
defaultReducers[sessionReducer] = session;
defaultReducers[gateKeeperReducerName] = gateKeeperReducer;

/** The initial store for the reveal web app */
export const store = getConnectedStore(defaultReducers, preloadedState);

/** Set listener to add reducers to store when registered */
reducerRegistry.setChangeListener((reducers) => {
  store.replaceReducer(combine(reducers));
});
