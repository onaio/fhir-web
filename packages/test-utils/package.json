{
  "name": "@opensrp/test-utils",
  "version": "0.0.0",
  "description": "",
  "main": "dist/index.js",
  "types": "dist/types",
  "files": [
    "dist"
  ],
  "publishConfig": {
    "access": "public"
  },
  "repository": "https://github.com/opensrp/web",
  "scripts": {
    "test": "run -T test $INIT_CWD --verbose",
    "tsc": "run -T tsc",
    "lint": "run -T eslint ./**/*.{js,jsx,ts,tsx}",
    "copy": "run -T copyfiles -u 1 \"./src/**/*.{css,html}\" \"./dist/\"",
    "build": "run tsc && run transpile && run copy",
    "transpile": "run -T babel src -d dist --root-mode upward --extensions .ts,.tsx --ignore '**/*.test.ts,**/*.test.tsx,**/tests,**/__tests__'"
  },
  "jest": {
    "automock": false,
    "setupFiles": [
      "../../setupTests"
    ]
  },
  "bugs": {
    "url": "https://github.com/opensrp/web/issues"
  },
  "dependencies": {
    "@onaio/connected-reducer-registry": "^0.0.3",
    "@opensrp/pkg-config": "^0.0.9",
    "@opensrp/rbac": "workspace:^"
  },
  "peerDependencies": {
    "@opensrp/store": "^0.0.10",
    "@testing-library/react": "^14.1.2",
    "@testing-library/user-event": "^13.5.0",
    "react": "^17.0.0 ",
    "react-redux": "^7.2.1",
    "react-router": "^5.2.1",
    "react-router-dom": "^5.2.1"
  },
  "author": "OpenSRP Engineering",
  "license": "Apache-2.0"
}
