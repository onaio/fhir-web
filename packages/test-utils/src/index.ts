export * from './componentWrappers';
export * from './utils'