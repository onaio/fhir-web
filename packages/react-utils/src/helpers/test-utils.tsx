import { store } from '@opensrp/store';
import { Provider } from 'react-redux';
import React from 'react';
import { RoleContext, UserRole, FhirResources, Permit, IamResources } from '@opensrp/rbac';
import { history } from '@onaio/connected-reducer-registry';
import { Router } from 'react-router';

export const superUserRole = new UserRole([...FhirResources, ...IamResources], Permit.MANAGE);

export interface ContextProviderProps {
  children: JSX.Element;
}

export const ContextProvider = (props: ContextProviderProps) => {
  const { children } = props;
  return (
    <Provider store={store}>
      <Router history={history}>
        <RoleContext.Provider value={superUserRole}>{children}</RoleContext.Provider>
      </Router>
    </Provider>
  );
};
