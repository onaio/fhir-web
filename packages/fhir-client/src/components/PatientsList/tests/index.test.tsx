import React from 'react';
import { Provider } from 'react-redux';
import { PatientsList } from '..';
import { Router } from 'react-router';
import { createBrowserHistory } from 'history';
import flushPromises from 'flush-promises';
import * as reactQuery from 'react-query';
import { act } from 'react-dom/test-utils';
import { store } from '@opensrp/store';
import { patients } from './fixtures';
import { mount, shallow } from 'enzyme';
import * as fhirCient from 'fhirclient';
import toJson from 'enzyme-to-json';

const { QueryClient, QueryClientProvider } = reactQuery;

const queryClient = new QueryClient();

jest.mock('@opensrp/notifications', () => ({
  __esModule: true,
  ...Object.assign({}, jest.requireActual('@opensrp/notifications')),
}));

const history = createBrowserHistory();

describe('Patients list view', () => {
  afterEach(() => {
    jest.resetAllMocks();
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });

  it('renders patients table without crashing', async () => {
    shallow(
      <Router history={history}>
        <QueryClientProvider client={queryClient}>
          <PatientsList fhirBaseURL="https://r4.smarthealthit.org/" />
        </QueryClientProvider>
      </Router>
    );
  });

  it('renders correctly', async () => {
    const fhir = jest.spyOn(fhirCient, 'client');
    fhir.mockImplementation(
      jest.fn().mockImplementation(() => {
        return {
          request: jest.fn().mockResolvedValueOnce(patients),
        };
      })
    );
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <QueryClientProvider client={queryClient}>
            <PatientsList fhirBaseURL="https://r4.smarthealthit.org/" />
          </QueryClientProvider>
        </Router>
      </Provider>
    );
    expect(toJson(wrapper.find('.ant-spin'))).toBeTruthy();

    await act(async () => {
      await flushPromises();
      wrapper.update();
    });
    expect(toJson(wrapper.find('.ant-spin'))).toBeFalsy();
    expect(wrapper.text()).toMatchSnapshot();

    // test sorting
    // find table (sorted by default order)
    wrapper.find('tr').forEach((tr, index) => {
      expect(tr.text()).toMatchSnapshot(`table rows - default order ${index}`);
    });

    // sort by username
    // click on sort to change the order (ascending)
    wrapper.find('thead tr th').first().simulate('click');
    wrapper.update();

    // check new sort order by name (ascending)
    wrapper.find('tr').forEach((tr, index) => {
      expect(tr.text()).toMatchSnapshot(`sorted table rows by name - ascending ${index}`);
    });

    // click on sort to change the order (descending)
    wrapper.find('thead tr th').first().simulate('click');
    wrapper.update();

    // check new sort order by name (ascending)
    wrapper.find('tr').forEach((tr, index) => {
      expect(tr.text()).toMatchSnapshot(`sorted table rows by name - descending ${index}`);
    });

    // cancel sort
    // click on sort to change the order (descending)
    wrapper.find('thead tr th').first().simulate('click');
    wrapper.update();

    // sort by dob
    // click on sort to change the order (ascending)
    expect(wrapper.find('thead tr th').at(1).text()).toEqual('Date Of Birth');
    wrapper.find('thead tr th').at(1).simulate('click');
    wrapper.update();
    // check new sort order by dob (ascending)
    wrapper.find('tr').forEach((tr, index) => {
      expect(tr.text()).toMatchSnapshot(`sorted table rows by dob - ascending ${index}`);
    });

    // click on sort to change the order (descending)
    wrapper.find('thead tr th').at(1).simulate('click');
    wrapper.update();

    // check new sort order by email (descending)
    wrapper.find('tr').forEach((tr, index) => {
      expect(tr.text()).toMatchSnapshot(`sorted table rows by dob - descending ${index}`);
    });
    wrapper.update();
    // sort by gender
    // click on sort to change the order (ascending)
    wrapper.find('thead tr th').at(2).simulate('click');
    wrapper.update();

    // check new sort order by name (ascending)
    wrapper.find('tr').forEach((tr, index) => {
      expect(tr.text()).toMatchSnapshot(`sorted table rows by gender - ascending ${index}`);
    });

    // click on sort to change the order (descending)
    wrapper.find('thead tr th').at(2).simulate('click');
    wrapper.update();

    // check new sort order by name (ascending)
    wrapper.find('tr').forEach((tr, index) => {
      expect(tr.text()).toMatchSnapshot(`sorted table rows by gender - descending ${index}`);
    });

    // cancel sort
    // click on sort to change the order (descending)
    wrapper.find('thead tr th').at(2).simulate('click');
    wrapper.unmount();
  });

  it('shows broken page if fhir api is down', async () => {
    const reactQueryMock = jest.spyOn(reactQuery, 'useQuery');
    reactQueryMock.mockImplementation(
      () =>
        ({
          data: undefined,
          error: 'Something went wrong',
          // eslint-disable-next-line @typescript-eslint/no-explicit-any
        } as any)
    );
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <QueryClientProvider client={queryClient}>
            <PatientsList fhirBaseURL="https://r4.smarthealthit.org/" />
          </QueryClientProvider>
        </Router>
      </Provider>
    );
    await act(async () => {
      await flushPromises();
    });
    wrapper.update();
    /** error view */
    expect(wrapper.text()).toMatchInlineSnapshot(`"ErrorSomething went wrongGo backGo home"`);
    wrapper.unmount();
  });
});
