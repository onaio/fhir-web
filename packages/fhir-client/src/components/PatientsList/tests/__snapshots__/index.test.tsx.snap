// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Patients list view renders correctly 1`] = `"PatientsAdd patientNameDate Of BirthGenderActions Apple Park  1988-08-04maleView Jane Doe  1988-08-04maleView John Doe  1988-08-04maleView Albert Einstein Deceased 1988-08-04maleView120 / page"`;

exports[`Patients list view renders correctly: sorted table rows by dob - ascending 0 1`] = `"NameDate Of BirthGenderActions"`;

exports[`Patients list view renders correctly: sorted table rows by dob - ascending 1 1`] = `" Albert Einstein Deceased 1988-08-04maleView"`;

exports[`Patients list view renders correctly: sorted table rows by dob - ascending 2 1`] = `" John Doe  1988-08-04maleView"`;

exports[`Patients list view renders correctly: sorted table rows by dob - ascending 3 1`] = `" Jane Doe  1988-08-04maleView"`;

exports[`Patients list view renders correctly: sorted table rows by dob - ascending 4 1`] = `" Apple Park  1988-08-04maleView"`;

exports[`Patients list view renders correctly: sorted table rows by dob - descending 0 1`] = `"NameDate Of BirthGenderActions"`;

exports[`Patients list view renders correctly: sorted table rows by dob - descending 1 1`] = `" Apple Park  1988-08-04maleView"`;

exports[`Patients list view renders correctly: sorted table rows by dob - descending 2 1`] = `" Jane Doe  1988-08-04maleView"`;

exports[`Patients list view renders correctly: sorted table rows by dob - descending 3 1`] = `" John Doe  1988-08-04maleView"`;

exports[`Patients list view renders correctly: sorted table rows by dob - descending 4 1`] = `" Albert Einstein Deceased 1988-08-04maleView"`;

exports[`Patients list view renders correctly: sorted table rows by gender - ascending 0 1`] = `"NameDate Of BirthGenderActions"`;

exports[`Patients list view renders correctly: sorted table rows by gender - ascending 1 1`] = `" Apple Park  1988-08-04maleView"`;

exports[`Patients list view renders correctly: sorted table rows by gender - ascending 2 1`] = `" Jane Doe  1988-08-04maleView"`;

exports[`Patients list view renders correctly: sorted table rows by gender - ascending 3 1`] = `" John Doe  1988-08-04maleView"`;

exports[`Patients list view renders correctly: sorted table rows by gender - ascending 4 1`] = `" Albert Einstein Deceased 1988-08-04maleView"`;

exports[`Patients list view renders correctly: sorted table rows by gender - descending 0 1`] = `"NameDate Of BirthGenderActions"`;

exports[`Patients list view renders correctly: sorted table rows by gender - descending 1 1`] = `" Albert Einstein Deceased 1988-08-04maleView"`;

exports[`Patients list view renders correctly: sorted table rows by gender - descending 2 1`] = `" John Doe  1988-08-04maleView"`;

exports[`Patients list view renders correctly: sorted table rows by gender - descending 3 1`] = `" Jane Doe  1988-08-04maleView"`;

exports[`Patients list view renders correctly: sorted table rows by gender - descending 4 1`] = `" Apple Park  1988-08-04maleView"`;

exports[`Patients list view renders correctly: sorted table rows by name - ascending 0 1`] = `"NameDate Of BirthGenderActions"`;

exports[`Patients list view renders correctly: sorted table rows by name - ascending 1 1`] = `" Albert Einstein Deceased 1988-08-04maleView"`;

exports[`Patients list view renders correctly: sorted table rows by name - ascending 2 1`] = `" Apple Park  1988-08-04maleView"`;

exports[`Patients list view renders correctly: sorted table rows by name - ascending 3 1`] = `" Jane Doe  1988-08-04maleView"`;

exports[`Patients list view renders correctly: sorted table rows by name - ascending 4 1`] = `" John Doe  1988-08-04maleView"`;

exports[`Patients list view renders correctly: sorted table rows by name - descending 0 1`] = `"NameDate Of BirthGenderActions"`;

exports[`Patients list view renders correctly: sorted table rows by name - descending 1 1`] = `" John Doe  1988-08-04maleView"`;

exports[`Patients list view renders correctly: sorted table rows by name - descending 2 1`] = `" Jane Doe  1988-08-04maleView"`;

exports[`Patients list view renders correctly: sorted table rows by name - descending 3 1`] = `" Apple Park  1988-08-04maleView"`;

exports[`Patients list view renders correctly: sorted table rows by name - descending 4 1`] = `" Albert Einstein Deceased 1988-08-04maleView"`;

exports[`Patients list view renders correctly: table rows - default order 0 1`] = `"NameDate Of BirthGenderActions"`;

exports[`Patients list view renders correctly: table rows - default order 1 1`] = `" Apple Park  1988-08-04maleView"`;

exports[`Patients list view renders correctly: table rows - default order 2 1`] = `" Jane Doe  1988-08-04maleView"`;

exports[`Patients list view renders correctly: table rows - default order 3 1`] = `" John Doe  1988-08-04maleView"`;

exports[`Patients list view renders correctly: table rows - default order 4 1`] = `" Albert Einstein Deceased 1988-08-04maleView"`;
